### PR TITLE
Multisite - Load customer data for logged in users regardless of being member of sub-site

### PR DESCRIPTION
**Pull request by mikejolley**
This is an attempted fix of #21580

Before this patch, if you're logged in, but not a member of a site, WordPress still treats you as a logged in users in some circumstances:

- is_user_logged_in is true
- user has an ID
- admin bar shows (if WC is off)

Additonally, if you login or checkout, WooCommerce will add you as a user on a site automatically.

Therefore, I don't understand why we'd want to limit customer data being loaded if you're logged in, regardless of site.

I also see no reason why API calls cannot add you to a site automatically like checkout does.

I've made those tweaks in this PR. If you repeat the test instructions in #21580 you'll see your address on the account page on other networked sites, and won't see a fatal error.

### CODE REVIEW

**Review by peterfabian — APPROVED** [review]
I verified that the use case mentioned in #21580 now works correctly, i.e. address is shown and can be edited without fatal errors.

Also, your reasoning for this change, given how WP core handles user vs how it was handled in WC prior to this change, makes sense to me. 

However, there might be a reason why it was working differently, of course. Please take this with a grain of salt, as this is the first time I've tried to set up multisite WP installation and I'm not very familiar how code should treat logged in users when they register on one site or the other.

File: includes/admin/class-wc-admin-menus.php [hunk 1]
```diff
@@ -346,7 +346,7 @@ public function admin_bar_menus( $wp_admin_bar ) {
 		}
 
 		// Show only when the user is a member of this site, or they're a super admin.
-		if ( ! is_user_member_of_blog() && ! is_super_admin() ) {
+		if ( ! is_super_admin() ) {
```
Comment (rodrigoprimo) [inline includes/admin/class-wc-admin-menus.php]: A user that has access to the dashboard on a site but is not super admin on the network will stop seeing the "Visit store" link after this change. Was this change intentional?

**Review by rodrigoprimo — COMMENTED** [review]
I followed the test instructions in the issue and confirmed that this PR fixes the problem. I did not test the API changes but checked the code and they make sense to me.

I just found one minor issue related to the "Visit store" link that we might need to address before merging this PR.

I pushed a commit to this branch fixing PHPCS errors that Travis was complaining.

File: includes/admin/class-wc-admin-menus.php (hunk 1 above)
Comment (claudiulodro) [reply includes/admin/class-wc-admin-menus.php]: I can confirm this. I don't think the original logic needed changing in this case.

**Review by claudiulodro — CHANGES_REQUESTED** [review]
The changes make sense for the my-account page and the REST API, and work correctly in both places in my testing. 👍 

Just one small bug introduced that needs to be handled, then this is good-to-go.

File: includes/admin/class-wc-admin-menus.php (hunk 1 above)
Comment (claudiulodro) [reply includes/admin/class-wc-admin-menus.php]: Shop managers are mostly who it would affect
